pyfar/dsp/filter/gammatone.py

```diff
@@ -0,0 +1,529 @@
+
+        phase_factors = 1j / (slopes / np.abs(slopes))
+
+        return delays, phase_factors
+
+    def _get_gains(self):
+        """
+        Section 4 in Hohmann 2002 describes how to derive these values. This
+        is a direct Python port of the corresponding function in the AMT
+        toolbox `hohmann2002_process.m`.
+        """
+
+        # positive and negative center frequencies in the z-plane
+        z = np.atleast_2d(
+            np.exp(2j * np.pi * self.frequencies / self.sampling_rate)).T
+        z_conj = np.conjugate(z)
+
+        # calculate transfer function at all center frequencies for all bands
+        # (matrixes contain center frequencies along first dimension and
+        h_pos = (1 - np.atleast_2d(self._coefficients) / z)**(-4) * \
+            np.atleast_2d(self._normalizations)
+        h_neg = (1 - np.atleast_2d(self._coefficients) / z_conj)**(-4) * \
+            np.atleast_2d(self._normalizations)
+
+        # apply delay and phase correction
+        phase_factors = np.atleast_2d(self._phase_factors)
+        delays = np.atleast_2d(self._delays)
+        h_pos *= phase_factors * z**(-delays)
+        h_neg *= phase_factors * np.conjugate(z)**(-delays)
+
+        # combine positive and negative spectrum
+        h = (h_pos + np.conjugate(h_neg)) / 2
+
+        # iteratively find gains
+        gains = np.ones((self.n_bands, 1))
+        for ii in range(100):
+            h_fin = np.matmul(h, gains)
+            gains /= np.abs(h_fin)
+
+        return gains.flatten()
+
+    def filter(self, signal, reset_state=True):
+        """
+        Filter an input signal.
+
+        The filter output is a complex valued time signal, whose real and
+        imaginary part are returned separately.
+
+        - If the filter bank ist used for analysis purposes only, the imaginary
+          part is not required for further processing.
+        - If the filter bank is used for analysis and re-synthesis, any further
+          processing must be applied to the real and imaginary part. Any
+          complex-valued operations must be applied to the complex valued
+          output as a whole.
+
+        Parameters
+        ----------
+        signal : Signal
+            The data to be filtered
+        reset_state : bool, optional
+            If true the internal state of the filter bank is reset before the
+            filters are applied. Not resetting the state can be useful for
+            blockwise processing. The default is ``True``.
+
+        Returns
+        -------
+        real : Signal
+            The real part of the complex output signal. This represents the
+            band-limited Gammatone filter output.
+        imag : Signal
+            The imaginary part of the complex output signal. This approximates
+            the Hilbert transform of the band-limited Gammatone filter output.
+
+        Notes
+        -----
+        - :math:`\\sqrt{real^2 + imag^2}` gives the envelope of the Gammatone
```

Comment:
I'd personally prefer not italic font here, but really important...

Reply:
I think that was the standard math font. In any way I converted it to code to be more explicit.